mlx/ops.cpp

```diff
@@ -2222,6 +2222,176 @@ array matmul(
   return out;
 }
 
+/** Computes: O = softmax(Q @ K.T) @ V; with parallelization across K-V sequence
+ * length **/
+array fast_inference_sdpa(
+    const array& queries,
+    const array& keys,
+    const array& values,
+    const float scale,
+    const std::optional<array>& mask,
+    StreamOrDevice s) {
+  auto q = queries;
+  auto k = keys;
+  auto v = values;
+  auto m = mask;
+
+  if (q.ndim() == 0 || k.ndim() == 0 || v.ndim() == 0) {
+    throw std::invalid_argument(
+        "[fast_inference_sdpa] Got 0 dimension input. Inputs must "
+        "have at least one dimension.");
+  }
```

Comment:
It seems like you check the input dims are not 0 here but then check again that they are exactly 4 below? Maybe this check is superfluous?

Reply:
yes, can remove the zero case and leave the rank 4 check